test/modules/layers/test_transformer.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+from test.test_utils import set_rng_seed, assert_expected
+from torchmultimodal.modules.layers.transformer import TransformerEncoder
+
+
+class TestTransformer(unittest.TestCase):
+    """
+    Test the Transformer classes
+    """
+
+    def setUp(self):
+        set_rng_seed(4)
+        self.encoder = TransformerEncoder(
+            hidden_size=2, num_attention_heads=2, num_hidden_layers=2
+        )
+        self.test_input = torch.rand((2, 3, 2))
+
+    def test_encoder(self):
+        output = self.encoder(self.test_input)
+
+        actual_last_hidden_state = output.last_hidden_state
+        actual_hidden_states = torch.stack(output.hidden_states)
+        actual_attentions = torch.stack(output.attentions)
+
+        expected_last_hidden_state = torch.Tensor(
+            [
+                [[0.4387, 2.2609], [0.4937, 2.1976], [0.1847, 2.4323]],
+                [[0.4651, 2.1418], [0.0404, 2.1412], [-0.1759, 1.9571]],
+            ]
+        )
+        expected_hidden_states = torch.Tensor(
+            [
+                [
+                    [[0.5924, 0.9998], [0.7723, 0.3792], [0.4945, 0.6260]],
+                    [[0.8161, 0.2282], [0.3914, 0.2276], [0.1751, 0.0436]],
+                ],
+                [
+                    [[0.7162, 1.0436], [0.7712, 0.9802], [0.4622, 1.2150]],
+                    [[0.7426, 0.9244], [0.3179, 0.9238], [0.1016, 0.7398]],
+                ],
+                [
+                    [[0.4387, 2.2609], [0.4937, 2.1976], [0.1847, 2.4323]],
+                    [[0.4651, 2.1418], [0.0404, 2.1412], [-0.1759, 1.9571]],
+                ],
+            ]
+        )
+        expected_attentions = torch.Tensor(
```

Comment:
Usually it is better to check for sum of the tensors to keep the code manageable.

Reply:
@ankitade should I go back to reducing the tensor completely or keep it like this as we discussed earlier?

Reply:
Its fine to leave them here. Its good to have granular checking at this point for guardrails for refactoring